polygeist/tools/cgeist/Lib/ValueCategory.cc

```diff
@@ -632,3 +658,15 @@ ValueCategory ValueCategory::InBoundsGEPOrSubIndex(OpBuilder &Builder,
                                                    ValueRange IdxList) const {
   return GEPOrSubIndex(Builder, Loc, Type, IdxList, /*IsInBounds*/ true);
 }
+
+template <typename OpTy>
+ValueCategory FPUnaryOp(OpBuilder &Builder, Location Loc, Value Val) {
+  assert(mlirclang::isFPOrFPVectorTy(Val.getType()) &&
+         "Expecting FP or FP vector operand type");
+  warnUnconstrainedOp<arith::NegFOp>();
+  return {Builder.createOrFold<OpTy>(Loc, Val), false};
+}
+
+ValueCategory ValueCategory::FNeg(OpBuilder &Builder, Location Loc) const {
```

Comment:
Can we pass the value to negate in the parameter list (like it is passed for FPUnaryOp) pls.

Reply:
This is a member function. Following the convention of the rest of the functions, we do not pass the "first" argument.